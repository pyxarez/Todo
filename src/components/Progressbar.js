import React from 'react';

import './Progressbar.res/styles.css';

Progressbar.propTypes = {
  progress: React.PropTypes.string
}


export default function Progressbar({ progress }) {
  return (
    <div className="my-progressbar-component">
      <div
        style={ progress && { width: progress }}
        className={"progressbar__state" + (progress === '100%' ? ' progressbar__state_full' : '')}/>
    </div>
  );
}
