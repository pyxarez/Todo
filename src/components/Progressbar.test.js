import React from 'react';
import { shallow } from 'enzyme';

import Progressbar from './Progressbar';

const setup = () => {
  const props = {
    progress: '20%'
  }
  const wrapper = shallow(<Progressbar {...props}/>)

  return {
    props,
    wrapper
  }
}

describe('Components :: Progressbar', () => {
  it('should render self', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});