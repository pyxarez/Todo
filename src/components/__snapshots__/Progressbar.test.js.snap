exports[`Components :: Progressbar should render self 1`] = `
<div
  className="my-progressbar-component">
  <div
    className="progressbar__state"
    style={
      Object {
        "width": "20%",
      }
    } />
</div>
`;
